Tools/Scripts/webkitpy/layout_tests/controllers/single_test_runner.py

```diff
@@ -222,13 +222,16 @@ def _save_baseline_data(self, data, extension, location):
 
         fs.maybe_make_directory(output_dir)
 
-        ext_parts = fs.splitext(self._test_name)
-
-        output_basename = fs.basename(ext_parts[0])
-        if len(ext_parts) > 1 and '?' in ext_parts[1]:
-            output_basename += '_' + ext_parts[1].split('?')[1]
-        if len(ext_parts) > 1 and '#' in ext_parts[1]:
-            output_basename += '_' + ext_parts[1].split('#')[1]
+        output_basename = fs.basename(self._test_name)
+        variant = ''
+        if '?' in output_basename:
+            (output_basename, variant) = output_basename.split('?')
```

Comment:
Should use `.split('?', 1)`, even though having a `?` in a variant would kind of be crazy.